### DISCUSSION TITLE
Hardware PIN warning after latest firmware dev20241002 on M5Stack Core2

### DISCUSSION THREAD

**Discussion by WH3080** [dl9rdz/rdz_ttgo_sonde#474] [Q&A] [answered]
Hello RDZsonde community, I apologize in advance for another newbie concern. I just installed the latest firmware via the the Espressif flash tool. I have not changed any hardware settings but get this warning:

"Duplicated PIN assignments found:
Pin 127 in 'Button input port' and 'Button 2 input port'
Pin 23 in 'OLED SDA/TFT SDA' and 'SX1278 MOSI'
Pin 18 in 'OLED SCL/TFT CLK' and 'SX1278 SCK'
Do you want to submit the form anyway?"

Now my (probably related) questions:
a) What are the recommended (hardware) settings for the M5 Core2?
b) Does the latest firmware still support the classic two hardware buttons on the M5 Core2
c) Does the latest firmware in addition / alternatively support the three software buttons (under red circles) on the M5 Core2 TFT touchscreen?
d) How do I set up the M5 correctly for the software buttons (if that is actually possible)?

Thank you in advance for your time and patience.

WH3080

**Comment by dl9rdz** [comment, marked as answer]
Autoconfiguration for the M5 Core2 should work.

The warning can be ignored (while all other boards use different PINs for the SX1278 and for other parts, it is a shared bus on the M5 Core2. The software handles this correctly internally, it is just the check not considering this case. This would be a configuration error on any other board)

The two "hardware" buttons on the Core2 should still work? (Not at home, can't check. Maybe one is just a reset button, so no software use for that, the other is a button handled by the AXP192, and this one can be used)

The touchscreen buttons are still not supported. This is on my low-priority todo list (if I have time some time....)

The SD card is not yet fully supported (unlike other boards, the SD card and the TFT display are on the same bus. This needs some better coordination in software). This is also on my (high-priority) todo list...

**Reply by WH3080** [reply]
Thank you DL9RDZ for the warp speed answer (and more importantly for dedicating your precious time to this project). Kindly, WH3080
